drivers/flash/spi_nor.c

```diff
@@ -438,8 +478,10 @@ static int read_sfdp(const struct device *const dev,
 static int enter_dpd(const struct device *const dev)
 {
 	int ret = 0;
+	const struct spi_nor_config *cfg = dev->config;
 
-	if (IS_ENABLED(DT_INST_PROP(0, has_dpd))) {
+	ARG_UNUSED(cfg);
+	if (IS_ENABLED(DT_INST_PROP(cfg->index, has_dpd))) {
```

Comment:
```
define DT_DRV_COMPAT jedec_spi_nor
#define DT_INST_PROP(inst, prop) DT_PROP(DT_DRV_INST(inst), prop)
#define DT_DRV_INST(inst) DT_INST(inst, DT_DRV_COMPAT)
#define DT_PROP(node_id, prop) DT_CAT3(node_id, _P_, prop)
#define DT_INST(inst, compat) UTIL_CAT(DT_N_INST, DT_DASH(inst, compat))
#define DT_DASH(...) MACRO_MAP_CAT(DT_DASH_PREFIX, __VA_ARGS__)
#define DT_DASH_PREFIX(name) _##name

DT_INST_PROP(cfg->index, has_dpd)->
DT_PROP(DT_DRV_INST(cfg->index, has_dpd)->
DT_PROP(DT_INST(cfg->index, DT_DRV_COMPAT ), has_dpd)->
DT_PROP(DT_INST(cfg->index, jedec_spi_nor), has_dpd)->
DT_PROP(UTIL_CAT(DT_N_INST, DT_DASH(cfg->index, jedec_spi_nor), has_dpd)->
DT_PROP(UTIL_CAT(DT_N_INST, _cfg->index_jedec_spi_nor), has_dpd)->
DT_PROP(DT_N_INST_cfg->index_jedec_spi_nor, has_dpd)

```

I do not think this compiles.

Reply:
Thank you for your feedback. I don't understand your comment. Can you please elaborate more? Why it doesn't compile?

Reply:
I'm sorry but I don't agree to the conclusion that it doesn't build.
From the zephyr documentation: 
[] (https://docs.zephyrproject.org/apidoc/latest/group__devicetree-generic-id.html#gae199b930cb21ff38dab284a696093ead )

> instance numbers have the following properties:
> 

- > for each compatible, instance numbers start at 0 and are contiguous

- > exactly one instance number is assigned for each node with a compatible, including disabled nodes

- > enabled nodes (status property is okay or missing) are assigned the instance numbers starting from 0, and disabled nodes have instance numbers which are greater than those of any enabled node

The process is:

1. Define multiple spi-nor devices in the device tree
2. The device instance numbers are generated, starting from 0
3. The configuration of each instance is done through the call of DT_INST_FOREACH_STATUS_OKAY(SPI_NOR_INST) 
4. Part of the configuration is to save the instance number in the index field of the spi_nor_config struct with the macro INST_CONFIG_STRUCT_GEN
5. cfg->index is used afterwards to differentiate the various instances of the config structure.

 I might have missed something, but it should work as it's based on what is done in other drivers as well

Reply:
Okay. I understand my mistake now. Any suggestion on how to handle this correctly?